### PR TITLE
Cows no longer spawn over water.

### LINKED ISSUE #2080: Cows spawn in oceans
It should probably be squids and other sea mobs, but when I fly across large ocean biomes, I see the server spawning groups of cows all over the place, instead.

### PR DESCRIPTION
**Pull request by bearbin**
Fixes #2080

### CODE REVIEW

**Comment by NiLSPACE** [comment]
Maybe also rabbits?

**Comment by bearbin** [comment]
Fixed.

File: src/Generating/FinishGen.cpp [hunk 1]
```diff
@@ -1259,7 +1259,7 @@ bool cFinishGenPassiveMobs::TrySpawnAnimals(cChunkDesc & a_ChunkDesc, int a_RelX
 	}
 	if (
 		(BlockUnderFeet != E_BLOCK_GRASS) &&
-		((AnimalToSpawn == mtSheep) || (AnimalToSpawn == mtChicken) || (AnimalToSpawn == mtPig))
+		((AnimalToSpawn == mtRabbit) || (AnimalToSpawn == mtCow) || (AnimalToSpawn == mtSheep) || (AnimalToSpawn == mtChicken) || (AnimalToSpawn == mtPig))
```
Comment (worktycho) [inline src/Generating/FinishGen.cpp]: This is getting very long. Perhaps this needs to be refactored out into a properties file similar to BlockInfo?

Comment (NiLSPACE) [inline src/Generating/FinishGen.cpp]: Maybe some code from the mobspawner can be shared? (not the blockentity, but the global mob spawner https://github.com/mc-server/MCServer/blob/master/src/MobSpawner.cpp)

Comment (worktycho) [inline src/Generating/FinishGen.cpp]: That seems like a good idea, but I would probably refactor that method out a bit. Its ridiculously long.

Comment (bearbin) [inline src/Generating/FinishGen.cpp]: You could refactor it into a case statement. Would that be good?

Comment (worktycho) [inline src/Generating/FinishGen.cpp]: I was referring to MobSpawner::CanSpawnHere which is 200 lines long.

Comment (bearbin) [inline src/Generating/FinishGen.cpp]: Ah ok. Sharing some code might be good, and refactoring would certainly be a good idea.

**Comment by bearbin** [comment]
Can this be merged, and another issue opened to refactor both methods?

**Pull request merged**